ufs2arco/regrid_ufs.py

```diff
@@ -0,0 +1,117 @@
+import yaml
+import numpy as np
+import xarray as xr
+from abc import ABC, abstractmethod
+
+from .gaussian_grid import gaussian_latitudes
+
+try:
+    import xesmf as xe
+except ImportError:
+    raise ImportError(
+        f"Cannot import xesmf. Install with 'conda install -c conda-forge xesmf',"
+        f" and note the environment must be deactivated and reactivated"
+    )
+
+
+class RegridUFS(ABC):
+    """
+    Regrid a UFS dataset
+
+    Example:
+
+        Construct the output grid. The following static methods are provided in this class
+            - compute_gaussian_grid
+            - compute_latlon_grid
+            - read_grid
+
+        >>> lats, lons = RegridUFS.compute_gaussian_grid(180, 360)
+
+        Open cice dataset using xarray or preferably :meth:`CICE6Dataset.open_dataset`
+
+        >>> ds = xr.open_mfdataset("./input/{ocn_,iceh_}*.nc")
+
+        Construct RegridCICE6 object, specifying output grid lats & lons, dataset and config file
+
+        >>> rg = RegridUFS(lats, lons, ds, config_filename = "config-replay.yaml")
+
+        Regrid dataset
+
+        >>> ds = rg.regrid(ds)
+    """
+
+    def __init__(
+        self,
+        lats1d_out: np.array,
+        lons1d_out: np.array,
+        ds_in: xr.Dataset,
+        config_filename: str,
+        interp_method: str = "bilinear",
+    ) -> None:
+        """
+        Initialize the RegridCICE6 object.
+        Args:
+            lats1d_out (array_like): One-dimensional array containing latitudes of the output grid.
+            lons1d_out (array_like): One-dimensional array containing longitudes of the output grid.
+            config_filename (str): YAML config file specifying options for regridding.
+            ds_in (xarray.Dataset): CICE6 xarray Dataset
+            interp_method (str, optional): Type of interpolation, default="bilinear".
```

Comment:
In the other UFSDataset classes I've been putting the parameter arguments inside of the class docstring. From what I've read online, this is apparently standard, so I'll suggest we keep to that for everything unless we find it to be disadvantageous later on.

```suggestion
```